panic in sources.(*kubeNodeMetrics).updateStats

heapster custom build from commit rev 20f9a2eac6f64d305a71d3c392ee9209c2e65e91
kubernetes : v1.0.5

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x68 pc=0x50a016]

goroutine 570 [running]:
k8s.io/heapster/sources.(*kubeNodeMetrics).updateStats(0xc2080e4a80, 0xc20831d5f0, 0x2f, 0xc2085fc300, 0xe, 0xc2087fe1f0, 0xe, 0xc20831a030, 0x2f, 0x5dc0, ...)
	/var/builds/workspace/46403-v3-component/BUILD_CONTAINER/docker/label/DOCKER-HIGH/app_root/tmp/go/src/k8s.io/heapster/sources/kube_nodes.go:86 +0xc06
k8s.io/heapster/sources.func·002(0xc20831d5f0, 0x2f, 0xc2085fc300, 0xe, 0xc2087fe1f0, 0xe, 0xc20831a030, 0x2f, 0x5dc0, 0xba87cf000)
	/var/builds/workspace/46403-v3-component/BUILD_CONTAINER/docker/label/DOCKER-HIGH/app_root/tmp/go/src/k8s.io/heapster/sources/kube_nodes.go:103 +0x10a
created by k8s.io/heapster/sources.(*kubeNodeMetrics).getNodesInfo
	/var/builds/workspace/46403-v3-component/BUILD_CONTAINER/docker/label/DOCKER-HIGH/app_root/tmp/go/src/k8s.io/heapster/sources/kube_nodes.go:111 +0x4a3

https://github.com/kubernetes/heapster/blob/20f9a2eac6f64d305a71d3c392ee9209c2e65e91/sources/kube_nodes.go#L86
i have a few nodes in the cluster in NotReady status (kubelet down).

--- a/sources/kube_nodes.go
+++ b/sources/kube_nodes.go
@@ -84,11 +84,13 @@ func (self *kubeNodeMetrics) updateStats(host nodes.Host, info nodes.Info, start
 		hostCopy := containers[hostIndex]
 		hostContainer = &hostCopy
 		containers = append(containers[:hostIndex], containers[hostIndex+1:]...)
+		// This is temporary workaround for #399. To make unit consistent with cadvisor normalize to a conversion factor of 1024.
+		hostContainer.Spec.Cpu.Limit = info.CpuCapacity * 1024 / 1000
+		hostContainer.Spec.Memory.Limit = info.MemCapacity
+		return hostContainer, containers, nil
+	} else {
+		return nil, []api.Container{}, fmt.Errorf("Host container not found")
 	}
-	// This is temporary workaround for #399. To make unit consistent with cadvisor normalize to a conversion factor of 1024.
-	hostContainer.Spec.Cpu.Limit = info.CpuCapacity * 1024 / 1000
-	hostContainer.Spec.Memory.Limit = info.MemCapacity
-	return hostContainer, containers, nil
 }
 
 // Returns the host containers, non-Kubernetes containers, and an error (if any).

